AngularApp/src/app/diagnostic-data/services/telemetry/telemetry.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable, OnInit, Inject } from '@angular/core';
+import { TelemetryEventNames, ITelemetryProvider } from './telemetry.common';
+import { DIAGNOSTIC_DATA_CONFIG, DiagnosticDataConfig } from '../../config/diagnostic-data-config';
+import { AppInsightsTelemetryService } from './appinsights-telemetry.service';
+import { KustoTelemetryService } from './kusto-telemetry.service';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable()
+export class TelemetryService {
+    private telemetryProviders: ITelemetryProvider[] = [];
+    private commonDetectorEventProperties: { [name: string]: string };
+    eventPropertiesSubject: BehaviorSubject<any> = new BehaviorSubject<any>(null);
+    eventPropertiesLocalCopy: { [name: string]: string };
```

Comment:
should this be private?

Reply:
Seems like we need to push data into eventPropertiesSubject from other components, like detector view component and generic detector component. There will be accessing issue if it's set as private. Is there any other way we can access this outside also assure its security?

Reply:
I meant about the localCopy.

Reply:
I see.. I have fixed that.